tools/Az.Tools.Predictor/Az.Tools.Predictor.Test/ParameterSetTests.cs

```diff
@@ -0,0 +1,278 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation.Language;
+using System.Management.Automation.Subsystem.Prediction;
+using Xunit;
+
+namespace Microsoft.Azure.PowerShell.Tools.AzPredictor.Test
+{
+    /// <summary>
+    /// Tests for <see cref="ParameterSet"/>. Positional parameter test cases requires the corresponding Az module installed.
+    /// </summary>
+    public sealed class ParameterSetTests : IDisposable
+    {
+        private AzContext _azContext;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="ParameterSetTests" />.
+        /// </summary>
+        public ParameterSetTests() => _azContext = new AzContext();
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            if (_azContext != null)
+            {
+                _azContext.Dispose();
+                _azContext = null;
+            }
+        }
+
+        /// <summary>
+        /// Verify that we can get the two named parameters.
+        /// </summary>
+        [Fact]
+        public void VerifyTwoCompleteNamedParameters()
+        {
+            var predictionContext = PredictionContext.Create("Get-AzResourceGroup -Name test -Location:WestUs2");
+            var commandAst = predictionContext.RelatedAsts.OfType<CommandAst>().LastOrDefault();
+            var expected = new List<Parameter>()
+            {
+                new Parameter("Name", "test", false),
+                new Parameter("Location", "WestUs2", false),
+            };
+
+            var parameterSet = new ParameterSet(commandAst, _azContext);
+            Assert.Equal(expected, parameterSet.Parameters);
+        }
+
+        /// <summary>
+        /// Verify that we can get the one named parameter.
+        /// </summary>
+        [Theory]
+        [InlineData("new-azresourcegroup -Name test")]
```

Comment:
Curious why this command is all lower case here?

Reply:
The PS commands are case insensitive. So I use lower cases for that case.